fission/src/systems/scene/GizmoSceneObject.ts

```diff
@@ -0,0 +1,194 @@
+                World.SceneRenderer.orbitControls.enabled = !event.value // disable orbit controls when dragging transform gizmo
+            }
+
+            if (event.target.mode === "translate") {
+                // disable other gizmos when translating
+                Array.from(World.SceneRenderer.sceneObjects.values())
+                    .filter(obj => obj instanceof GizmoSceneObject)
+                    .forEach(obj => {
+                        if (obj.gizmo.object === event.target.object && obj.gizmo.mode !== "translate") {
+                            obj.gizmo.dragging = false
+                            obj.gizmo.enabled = !event.value
+                            return
+                        }
+                    })
+            } else if (
+                event.target.mode === "scale" &&
+                (InputSystem.isKeyPressed("ShiftRight") || InputSystem.isKeyPressed("ShiftLeft"))
+            ) {
+                // scale uniformly if shift is pressed
+                event.target.axis = "XYZE"
+            } else if (event.target.mode === "rotate") {
+                // scale on all axes
+                Array.from(World.SceneRenderer.sceneObjects.values())
+                    .filter(obj => obj instanceof GizmoSceneObject)
+                    .forEach(obj => {
+                        if (
+                            obj.gizmo.mode === "scale" &&
+                            event.target !== obj.gizmo &&
+                            obj.gizmo.object === event.target.object
+                        ) {
+                            obj.gizmo.dragging = false
+                            obj.gizmo.enabled = !event.value
+                            return
+                        }
+                    })
+            }
+        })
+    }
+
+    public Update(): void {
+        this._gizmo.updateMatrixWorld()
+
+        // updating the size of the gizmo based on the distance from the camera
+        const mainCameraFovRadians = (Math.PI * (this._mainCamera.fov * 0.5)) / 180
+        this._gizmo.setSize(
+            (this._size / this._mainCamera.position.distanceTo(this.gizmo.object!.position)) *
+                Math.tan(mainCameraFovRadians) *
+                1.9
+        )
+
+        /** Translating the obj changes to the mirabuf scene object */
+        if (this._parentObject) {
+            this._parentObject.DisablePhysics()
+            if (this.isDragging) {
+                this._parentObject.mirabufInstance.parser.rigidNodes.forEach(rn => {
+                    this.UpdateBodyPositionAndRotation(rn)
+                    this._parentObject?.UpdateNodeParts(rn, this.obj.matrix)
+                })
+            }
+        }
+
+        // creating enter key and escape key event listeners
+        if (InputSystem.isKeyPressed("Enter") && this._parentObject) {
+            // confirming placement of object
+            if (this._parentObject !== undefined) this._parentObject.EnablePhysics()
```

Comment:
```suggestion
            if (this._parentObject) this._parentObject.EnablePhysics()
```

I know `this._parentObject` can only be a `MirabufSceneObject` or `undefined` but I still prefer a truthiness check rather than checking if it's not exactly undefined. It doesn't really matter very much in this instance though.